pkg/watcher/results/results.go

```diff
@@ -116,6 +120,25 @@ func (c *Client) ensureResult(ctx context.Context, o Object, opts ...grpc.CallOp
 		}
 	}
 
+	// Set the Result.Annotations and Result.Summary.Annotations fields if
+	// the object in question contains the required annotations.
+
+	if value, found := o.GetAnnotations()[annotation.ResultAnnotations]; found {
+		if annotations, err := parseAnnotations(annotation.ResultAnnotations, value); err != nil {
+			return nil, err
+		} else {
+			res.Annotations = annotations
```

Comment:
Are there other annotations that we add to the Result record? Should we merge annotations instead of overwrite?